packages/expo-router/types/expo-router.d.ts

```diff
@@ -0,0 +1,345 @@
+/* eslint-disable */
```

Comment:
This is a new centralised location for Expo Router types that can be shared by both the code-base and typed routes. After this is merged, I'd adjust the `expo-router` package to use these types.